src/actionCreators/resourceTemplates.js

```diff
@@ -13,53 +12,50 @@ import { findResourceTemplate } from 'selectors/entitySelectors'
 import _ from 'lodash'
 
 // A thunk that gets a resource template from state or the server.
-export const fetchResourceTemplate = resourceTemplateId => (dispatch, getState) => {
+export const fetchResourceTemplate = (resourceTemplateId, errorKey) => (dispatch, getState) => {
   // Try to get it from state.
   const resourceTemplate = findResourceTemplate(getState(), resourceTemplateId)
   if (resourceTemplate) return resourceTemplate
 
-  // Otherwise, retrieve from server and add to state.
-  dispatch(retrieveResourceTemplateStarted(resourceTemplateId))
-
   return getResourceTemplate(resourceTemplateId, 'ld4p').then((response) => {
     // If resource template loads, then validate.
     const resourceTemplate = response.response.body
-    return validateResourceTemplate(resourceTemplate).then((reason) => {
-      if (_.isEmpty(reason)) {
+    return validateResourceTemplate(resourceTemplate).then((errors) => {
+      if (_.isEmpty(errors)) {
         dispatch(setResourceTemplate(resourceTemplate))
         return resourceTemplate
       }
-      dispatch(setRetrieveResourceTemplateError(resourceTemplateId, reason))
+      errors.forEach(error => dispatch(appendError(errorKey, error)))
     }).catch((err) => {
       console.error(err)
-      dispatch(setRetrieveResourceTemplateError(resourceTemplateId, err.toString()))
+      dispatch(appendError(errorKey, `Error retrieving ${resourceTemplateId}: ${err.toString()}`))
```

Comment:
Instead of dispatching this error and writing it to the console, could you raise this as an error and the outer catch (starting on line 30) would handle the dispatch and console logging?

Reply:
Fixed.